src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Resource/FindAzureResourceCmdlet.cs

```diff
@@ -0,0 +1,492 @@
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.MultiSubscriptionListResourcesParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The extension resource type. e.g. Microsoft.Sql/Servers/Databases.")]
+        [ValidateNotNullOrEmpty]
+        public string ExtensionResourceType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the top parameter.
+        /// </summary>
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListResourcesParameterSet, Mandatory = false, HelpMessage = "The number of resources to retrieve.")]
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListTenantResourcesParameterSet, Mandatory = false, HelpMessage = "The number of resources to retrieve.")]
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.MultiSubscriptionListResourcesParameterSet, Mandatory = false, HelpMessage = "The number of resources to retrieve.")]
+        [ValidateNotNullOrEmpty]
+        public int? Top { get; set; }
+
+        /// <summary>
+        /// Gets or sets the OData query parameter.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "An OData style filter which will be appended to the request in addition to any other filters.")]
+        [ValidateNotNullOrEmpty]
+        public string ODataQuery { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource group name.
+        /// </summary>
+        [Alias("ResourceGroupName")]
+        [Parameter(Mandatory = false, ParameterSetName = FindAzureResourceCmdlet.ListResourcesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group name substring.")]
+        [Parameter(Mandatory = false, ParameterSetName = FindAzureResourceCmdlet.MultiSubscriptionListResourcesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group name substring.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupNameContains { get; set; }
+
+        /// <summary>
+        /// Gets or sets the expand properties property.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "When specified, expands the properties of the resource.")]
+        public SwitchParameter ExpandProperties { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tenant level parameter.
+        /// </summary>
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListTenantResourcesParameterSet, Mandatory = true, HelpMessage = "Indicates that this is a tenant level operation.")]
+        public SwitchParameter TenantLevel { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource property object format.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "The output format of the resource properties.")]
```

Comment:
What are the possible valid values for the output format - json, csv ? or something else